config/management/operational/src/network_checker.rs

```diff
@@ -182,187 +188,137 @@ impl CheckValidatorSetEndpoints {
         Err(Error::CommandArgumentError(
-            "Address must have ip, tcp, noise key, and handshake".to_string(),
+            "Address must have ip+tcp or dns+tcp, noise pubkey, and handshake".to_string(),
         ))
     } else {
         Ok(())
     }
 }
 
 /// Wrapper for `check_endpoint_inner` to handle runtime
-fn check_endpoint(
-    upgrade_context: Arc<UpgradeContext>,
+async fn check_endpoint(
+    transport: &DiemNetTransport<TcpTransport>,
     address: NetworkAddress,
-    timeout: Duration,
     no_handshake: bool,
 ) -> Result<String, Error> {
-    let runtime = Runtime::new().unwrap();
-    let remote_pubkey = address.find_noise_proto().unwrap();
-
-    let connect_future = async {
-        tokio::time::timeout(timeout, async {
-            if no_handshake {
-                check_endpoint_inner_no_handshake(address.clone()).await
-            } else {
-                check_endpoint_inner(upgrade_context.clone(), address.clone(), remote_pubkey).await
-            }
-        })
-        .await
-        .map_err(|_| Error::Timeout("CheckEndpoint", address.to_string()))
+    let result = if no_handshake {
+        check_endpoint_inner_no_handshake(address.clone()).await
+    } else {
+        check_endpoint_inner(transport, address.clone()).await
     };
-
-    runtime.block_on(connect_future)?
+    result.map_err(|err| {
+        Error::UnexpectedError(format!("Failed check for address {}: {:#}", address, err))
+    })
 }
 
-/// Connects via Noise, and then drops the connection
+/// Connects via the actual production DiemNetTransport. This requires a keypair
+/// that the remote will actually accept.
 async fn check_endpoint_inner(
-    upgrade_context: Arc<UpgradeContext>,
+    transport: &DiemNetTransport<TcpTransport>,
     address: NetworkAddress,
-    remote_pubkey: x25519::PublicKey,
-) -> Result<String, Error> {
-    // Connect to the address, this should handle DNS resolution
-    let fut_socket = async {
-        resolve_and_connect(address.clone())
-            .await
-            .map(TcpSocket::new)
-    };
-
+) -> Result<String> {
     // The peer id doesn't matter because we don't validate it
-    let remote_peer_id = account_address::from_identity_public_key(remote_pubkey);
-    match upgrade_outbound(
-        upgrade_context,
-        fut_socket,
-        address.clone(),
-        remote_peer_id,
-        remote_pubkey,
-    )
-    .await
-    {
-        Ok(conn) => {
-            let msg = format!("Successfully connected to {}", conn.metadata.addr);
-
-            // Disconnect
-            drop(conn);
-            Ok(msg)
-        }
-        Err(error) => Err(Error::UnexpectedError(format!(
-            "Failed to connect to {} due to {}",
-            address, error
-        ))),
-    }
+    let remote_peer_id = PeerId::ZERO;
+
+    let conn = transport
+        .dial(remote_peer_id, address)
+        .context("Failed to create dialer")?
+        .await
+        .context("Failed to connect")?;
```

Comment:
Will it now tell you why you the specific reason why it failed to connect still?  E.g. before it would tell you your chain Id didn't match etc which made it a useful debugging tool.